File: official/guides/getting-started/java/create_shipment.java
package getting_started;

import java.util.HashMap;

import com.easypost.EasyPost;
import com.easypost.exception.EasyPostException;

public class CreateShipment {
    public static void main(String[] args) throws EasyPostException {
        EasyPost.apiKey = System.getenv("EASYPOST_API_KEY");

        Map<String, Object> shipmentMap = new HashMap<String, Object>();
        shipmentMap.put("to_address", toAddress);
        shipmentMap.put("from_address", fromAddress);
        shipmentMap.put("parcel", parcel);

        Shipment shipment = Shipment.create(shipmentMap);
    }
}

File: official/landing_pages/label_api/create_shipment.rb
require 'easypost'

EasyPost.api_key = "<YOUR_TEST/PRODUCTION_API_KEY>"

to_address = EasyPost::Address.create(...)
from_address = EasyPost::Address.create(...)
parcel = EasyPost::Parcel.create(...)

shipment = EasyPost::Shipment.create(
  to_address: to_address,
  from_address: from_address,
  parcel: parcel,
  options: { label_format: 'ZPL' }
)
